firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -62,34 +67,49 @@ public class ConfigRealtimeHttpClient {
   private final Set<ConfigUpdateListener> listeners;
 
   @GuardedBy("this")
-  private HttpURLConnection httpURLConnection;
+  private Future<?> autoFetchTask;
+
+  @GuardedBy("this")
+  private int RETRIES_REMAINING;
+
+  @GuardedBy("this")
+  private long RETRY_TIME_SECONDS;
+
+  @GuardedBy("this")
```

Comment:
Instances of `Random` are thread safe https://developer.android.com/reference/java/util/Random

Reply:
Removed guardedby